File: server/middleware/logger.ts
// H3 middleware: logs method, URL, status, and response time for every request
export default defineEventHandler((event) => {
  const start = Date.now();
  const { method, url } = event.node.req;

  event.node.res.on("finish", () => {
    const ms = Date.now() - start;
    const status = event.node.res.statusCode;
    console.log(`${method} ${url} ${status} - ${ms}ms`);
  });
});

File: internal/delivery/http/middleware/logger.go
package middleware

import (
	"log"
	"time"

	"github.com/gin-gonic/gin"
)

func Logger() gin.HandlerFunc {
	return func(c *gin.Context) {
		start := time.Now()
		path := c.Request.URL.Path
		raw := c.Request.URL.RawQuery

		c.Next()

		timeStamp := time.Now().Format("2006-01-02T15:04:05.000Z07:00")
		latency := time.Since(start)
		statusCode := c.Writer.Status()
		clientIP := c.ClientIP()
		method := c.Request.Method

		if raw != "" {
			path = path + "?" + raw
		}

		log.Printf("%s - %s %s | %d | %s | %s", timeStamp, method, path, statusCode, latency, clientIP)
	}
}
